feat: add team groups section in group configurations

BryanttV opened the pull request:
## Description

This PR adds a new section (`TeamGroupsSection`) to the Group Configurations page. This section is necessary to render the **Teams** configured in each **Group** of the course when the functionality to connect Teams with Content Groups is enabled.

## Supporting information

- openedx/frontend-app-authoring/issues/1698
- [Backend Implementation](https://github.com/openedx/edx-platform/pull/33788)
- [Product Proposal](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3885662226/Connecting+Teams+to+Content+Groups)

## Screenshots

### Before
![image](https://github.com/user-attachments/assets/fc0bdbff-1996-4b94-953a-4fbbd259f26e)

### After
![image](https://github.com/user-attachments/assets/f7c06471-5602-4788-8df3-e59765bf19c6)

## Testing instructions

1. Create a Waffle Flag for your course in `{lms_domain}/admin/waffle_utils/waffleflagcourseoverridemodel/`

    - Waffle Flag: `teams.content_groups_for_teams`
    - Course id: [Your Course ID]
    - Enabled: ✅ 
2. Enables Teams in the platform using the `teams.enable_teams_app` flag.
3. Enable teams for your course in **Studio > [Your Course] > Content > Pages & Resources > Teams > Teams toggle**
4. In the same modal of **Configure teams**, create a few **Groups**.
5. Then, go to the **LMS > [Your Course] > Teams** and create a few **Teams** within those **Groups (Topics/team-sets)**
6. From Studio, go to **Settings > Group Configurations**.
7. You can see the new section for each of the groups. These sections are not editable.
8. Also, you can create **Content Groups** without any problem.

[add-team-groups-section-in-group-configurations.webm](https://github.com/user-attachments/assets/7e5dde1e-8e7c-475e-a4b5-2b323cbf70fb)

## Additional Information
This is how it looks in the legacy interface:

[teams-groups-legacy-interface.webm](https://github.com/user-attachments/assets/b3352f33-8da3-4f13-a3df-004836536c37)

openedx-webhooks commented:
Thanks for the pull request, @BryanttV!


This repository is currently maintained by `@openedx/2u-tnl`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

dcoa reviewed (approved):
LGTM!!

@bradenmacdonald  @PKulkoRaccoonGang I don't have permission over this repo, do you mind have a look to this PR?

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1728-49e51b.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1728-49e51b.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-9616468481-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-9616468481-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-9616468481-tutor-requirements.txt)

PKulkoRaccoonGang commented on src/group-configurations/team-groups-section/TeamGroupsSection.test.jsx:
```diff
@@ -0,0 +1,24 @@
+import { teamGroupsMock } from '../__mocks__';
+import TeamGroupsSection from '.';
+
+const renderComponent = (props = {}) => render(
```
[optional]: It seems that the [customRender utility](https://github.com/openedx/frontend-app-authoring/blob/master/src/testUtils.tsx#L140) should provide all the necessary wrappers for tests.

BryanttV replied:
Done: https://github.com/openedx/frontend-app-authoring/pull/1728/commits/5fe98985da5105b428b622d767ede8c29f1baeb6

PKulkoRaccoonGang commented on src/group-configurations/team-groups-section/index.jsx:
```diff
@@ -0,0 +1,23 @@
+import { availableGroupPropTypes } from '../constants';
+import ContentGroupCard from '../content-groups-section/ContentGroupCard';
+
+const TeamGroupsSection = ({ availableGroup: { groups, name } }) => (
```
[optional]: It would be great to add new components in Typescript

BryanttV replied:
Done!

PKulkoRaccoonGang reviewed (commented):
Thanks for the PR! After some testing, I have some suggestions/observations that probably don't apply to this PR. If they don't apply to your scope of work and they are valid, I don't mind moving them to a separate issue 💯 

1. If the course unit page is enabled, the link (which is under This group controls access to:) should be followed without reloading the page.
2. After I create a team I get an error with status code 500
![Screenshot 2025-04-04 at 12 12 59](https://github.com/user-attachments/assets/4b9dc9b7-de82-40e6-834c-89321f663928)
3. After adding a new group, the user must manually scroll the inner content of the Configure teams modal window to the newly added group. Should the modal window content be automatically scrolled to the new group block?
![image](https://github.com/user-attachments/assets/f46d6cdf-8d53-44b1-8626-2f9da904390a)
4. Is it worth showing the icon to collapse the new group creation block if it is not possible to do so until the group is created? I think this can be a bit misleading for users.
![image](https://github.com/user-attachments/assets/88aed4a1-f036-493a-8c25-dc328abe4ce8)

BryanttV commented:
Hi @PKulkoRaccoonGang! Thank you for your review, and I apologize for the delay. I was busy with other things. I've already addressed your suggestions.

Regarding what you found, I agree it should be moved to a separate issue.

FYI, the 500 error has already been reported here: https://github.com/openedx/edx-platform/issues/36841, and work is in progress to fix it.

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-10816534050.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-10816534050-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-10816534050-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-10816534050-tutor-requirements.txt)

bradenmacdonald commented:
@PKulkoRaccoonGang do you have time to resume your review here?

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11159925014.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11159925014-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11159925014-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11159925014-tutor-requirements.txt)

BryanttV commented:
@bradenmacdonald, thanks! rebased with master ✅

bradenmacdonald reviewed (changes requested):
When testing this, I found that I'm seeing some team groups listed twice. Let me know if you need more info to debug.

<img width="1476" height="1162" alt="Screenshot 2025-09-04 at 2 39 15 PM" src="https://github.com/user-attachments/assets/59ac52e4-5689-4001-99bd-fdedd4bbb9d1" />

The legacy UI only shows them once:

<img width="852" height="794" alt="Screenshot 2025-09-04 at 2 40 44 PM" src="https://github.com/user-attachments/assets/0a50c3bf-3d06-4032-9e93-15e3395fd70c" />

Also, I'm still unable to create a new content group. It gives the error shown below:

<img width="1328" height="616" alt="Screenshot 2025-09-04 at 2 41 46 PM" src="https://github.com/user-attachments/assets/3fd426b4-b11b-4d7d-b35f-72a8f0f96c9a" />

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252697.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252697-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252697-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252697-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1728-49e51b.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1728-49e51b.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252659-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252659-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-11904252659-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1728-49e51b.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1728-49e51b.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-12975970892-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-12975970892-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-12975970892-tutor-requirements.txt)

BryanttV commented on src/group-configurations/team-groups-section/TeamGroupsSection.test.jsx:
```diff
@@ -0,0 +1,24 @@
+import { initializeMocks, render } from '../../testUtils';
```
File updated, thanks!

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-13282907951.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-13282907951-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-13282907951-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1728-49e51b-51564656-13282907951-tutor-requirements.txt)
